### ISSUE TITLE
Batching images first or formatting them first?

### ISSUE THREAD

**Issue by ManishAradwad** [tensorflow/tensorflow#35816]
https://github.com/tensorflow/examples/blob/master/courses/udacity_intro_to_tensorflow_lite/tflite_c04_exercise_convert_model_to_tflite_solution.ipynb

Under "Create a Dataset from Images and Labels", we have this code

`train_batches=train_examples.cache().shuffle(num_examples//4).batch(BATCH_SIZE).map(format_example).prefetch(1)`

and similar for the validation and test examples...

Is this a right practise?? 
Shouldn't we first format the raw images and then batch them together rather than opposite way?

**Comment by ashutosh1919** [comment]
@ManishAradwad , I don't think that it is a problem to batch together first and pre-processing later. These two things are independent and if the data is well shuffled then we can batch them before. Please state any disadvantages of this implementation?

**Comment by ManishAradwad** [comment]
Hey @ashutosh1919 , sorry for the late reply...

I actually came across this issue while doing the exercise in the dl course. I've tried replicating the problem in following colab: https://github.com/ManishAradwad/ML_Stuff/blob/master/Courses/Udacity:%20Intro%20to%20Tensorflow%20for%20DeepLearning/Lesson5:%20TransferLearning/FlowersWithTransferLearning.ipynb

If we batch the flower images first and then map them, then we get following error while training-

`InvalidArgumentError: 2 root error(s) found.
(0) Invalid argument:  Cannot batch tensors with different shapes in component 0. First element had shape [213,320,3] and element 1 had shape [240,240,3].
	 [[node IteratorGetNext (defined at <ipython-input-40-b1d077d96724>:9) ]]
(1) Invalid argument:  Cannot batch tensors with different shapes in component 0. First element had shape [213,320,3] and element 1 had shape [240,240,3].
	 [[node IteratorGetNext (defined at <ipython-input-40-b1d077d96724>:9) ]]
	 [[IteratorGetNext/_2]]`

`0 successful operations.
0 derived errors ignored. [Op:__inference_distributed_function_147782]`

`Function call stack:
distributed_function -> distributed_function`

I think it has something to do with the number of color channels since the code for MNIST works flawlessly.
